gluten-core/src/main/scala/io/glutenproject/expression/ExpressionConverter.scala

```diff
@@ -275,6 +275,10 @@ object ExpressionConverter extends SQLConfHelper with Logging {
           cw
         )
       case i: In =>
+        if (i.list.exists(!_.foldable)) {
+          throw new UnsupportedOperationException(
+            s"In list option does not support non-foldable expression, ${i.list.map(_.sql)}")
+        }
```

Comment:
@zzcclp Is non-foldable expression not supported by CH backend either?

Reply:
yes, not supported.